src/faction_camp.cpp

```diff
@@ -535,23 +535,38 @@ static cata::optional<basecamp *> get_basecamp( npc &p, const std::string &camp_
 recipe_id base_camps::select_camp_option( const std::map<recipe_id, translation> &pos_options,
         const std::string &option )
 {
-    std::vector<recipe_id> pos_name_ids;
     std::vector<std::string> pos_names;
+    int choice = 0;
+
     for( const auto &it : pos_options ) {
         pos_names.push_back( it.second.translated() );
-        pos_name_ids.push_back( it.first );
     }
 
-    if( pos_name_ids.size() == 1 ) {
```

Comment:
I would actually hoist this to the very top of the function and not involve any helper containers:
```
if( pos_options.size() == 1 ) {
    return pos_options.front().first;
}
```
Then we proceed to handle the size > 1 case in the rest of the function.